backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -93,4 +94,13 @@ public GetShopsByShopOwnerResponse getShopsByShopOwner(String shopOwnerId) {
 							.collect(Collectors.toList());
 		return new GetShopsByShopOwnerResponse(shopOwnerMap.get(shopOwnerId), shops);
 	}
+
+	public DeleteTokenResponse deleteToken(UUID tokenId, Boolean isLoggedin) {
+		if (!isLoggedin) {
+			tokenMap.get(tokenId).setStatus(Status.CANCELLED_BY_CUSTOMER);
+		} else {
+			tokenMap.get(tokenId).setStatus(Status.CANCELLED_BY_SHOP_OWNER);
+		}
+		return new DeleteTokenResponse(tokenMap.get(tokenId));
+	}
```

Comment:
Are you supposed to change the status of the token to be deleted in the shopIdToListOfTokensMap too??

Reply:
did it tough I don't think we would need it as spring boot is based on resource sharing principle.

Reply:
Anyways I have added it to the code for now, for better clarity.